pkg/executor/infoschema_reader.go

```diff
@@ -210,6 +211,9 @@ func (e *memtableRetriever) retrieve(ctx context.Context, sctx sessionctx.Contex
 			return nil, err
 		}
 		e.initialized = true
+		if e.memTracker != nil {
+			e.memTracker.Consume(calculateDatumsSize(e.rows))
```

Comment:
At this time, the memory has already been allocated :thinking:. Do we have a plan to `Consume` progressively in each implementation above?